### PR TITLE
💫 Add interface generation

### LINKED ISSUE #2: 🌱Create Interface
Create `storer` Interface from REST definition. The following Sysl spec
```
RestApi [json_map_fields="dashes", interface="Storer", interface_doc="Storer abstracts all required RefData persistence and retrieval"]:
    /api:
        GET [method_name="GetKeys", method_doc="DataSet"]:
            return Keys
        POST (ds <: DataSetPayload) [method_name="CreateDataSet"]:
            return Key
    /api/{key<:string}:
        GET ?time={queryTime<:string} [method_name="GetData",method_doc="Data"]:
            return Data
        PUT(dp <: DataPayload) [method_name="PutData"]:
            return Data
```
should become
```go
// Storer abstracts all required RefData persistence and retrieval
type Storer interface {
	// DataSet
	GetKeys() (Keys, error)
	CreateDataSet(DataSetPayload) (Key, error)
	// Data
	GetData(key string, queryTime string) (Data, error)
	PutData(key string, dp DataPayload) (Data, error)
}
```

### PR DESCRIPTION
**Pull request by juliaogris**
### Proposed changes
Interace generation and tests, e.g.:
```go

// Storer abstracts all required RefData persistence and retrieval
type Storer interface {

	// DataSet
	GetKeys() (Keys, error)
	CreateDataSet(ds DataSetPayload) (Key, error)
	GetDataSetName(key string, queryTime string) (KeyName, error)
	PutDataSetName(key string, np NamePayload) (KeyName, error)

	// Data
	GetData(key string, queryTime string) (Data, error)
	PutData(key string, dp DataPayload) (Data, error)
	GetDataWithStart(startTime string, key string) (Data, error)
	PutDataWithStart(startTime string, key string, dp DataPayload) (Data, error)
	DeleteData(startTime string, key string) error

	// ...
}
```

### Fixes #2

### CODE REVIEW

File: interface.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+func genMethodName(ep *pb.Endpoint) string {
+	if n, ok := ep.Attrs["method_name"]; ok {
+		return n.GetS()
+	}
+	name := strings.Replace(ep.Name, " ", "", -1)
+	name = strings.Replace(name, "{", "", -1)
+	name = strings.Replace(name, "}", "", -1)
+
+	name = strings.Replace(name, "-", "/", -1)
```
Comment (marcelocantos) [inline interface.go]: Use `regexp` `ReplaceAllStringLiteral("[-_.,#]", …)`.

**Comment by juliaogris** [comment]
@anzdaddy - fixed, please have another look.

File: interface.go [hunk 2]
```diff
@@ -36,15 +36,12 @@ func genMethodName(ep *pb.Endpoint) string {
...
-	name := strings.Replace(ep.Name, " ", "", -1)
-	name = strings.Replace(name, "{", "", -1)
-	name = strings.Replace(name, "}", "", -1)
 
-	name = strings.Replace(name, "-", "/", -1)
-	name = strings.Replace(name, "_", "/", -1)
-	name = strings.Replace(name, ".", "/", -1)
-	name = strings.Replace(name, ",", "/", -1)
-	name = strings.Replace(name, "#", "/", -1)
+	re := regexp.MustCompile(`[{}\s]`)
```
Comment (marcelocantos) [inline interface.go]: Package-level var.

**Comment by juliaogris** [comment]
@anzdaddy, @marcelocantos done. Please have another look.

**Pull request merged**